packages/mui-material/src/Grid/Grid.js

```diff
@@ -193,7 +193,16 @@ const GridRoot = styled('div', {
       container && styles.container,
       item && styles.item,
       zeroMinWidth && styles.zeroMinWidth,
-      container && spacing !== 0 && styles[`spacing-xs-${String(spacing)}`],
+      container &&
+        spacing &&
+        spacing !== 0 &&
+        typeof +spacing === 'number' &&
```

Comment:
Should it rather be like this?
```suggestion
        typeof spacing === 'number' &&
```
because
```tsx
// always evaluates to `true`
typeof +spacing === 'number'
```

Reply:
I see my mistake (not the first time I’ve been tripped up by NaN) - this is the condition I was trying to write:

`isNaN(Number(spacing))`

It will only evaluate to true if spacing is already a number or if spacing can be converted to a valid number.

The spacing property can be a string, number, or object. Strings that can be converted to valid numbers should be accepted and processed as if they are numbers. Other strings should be ignored.

Reply:
@Janpot I added an additional commit that corrects this error.